lib/avalon/ffprobe.rb

```diff
@@ -14,10 +14,12 @@
 
 module Avalon
   class FFprobe
-    # media_path should be the output of `FileLocator.new(file_location).location`
-    def initialize(media_path)
+    attr_reader :json_output, :video_stream, :audio_stream
```

Comment:
A pattern I've seen before that I think would work well here is defining getter methods for these and using memoization.  I think that would also help cleanup the initializer method and make it lazily call ffprobe when one of these getters is called.

I'm thinking something like this:
```
  def initialize(media_file)
    @media_file = media_file
  end

  def json_output
    return @json_output unless @json_output.nil?
    return @json_output = {} unless valid_content_type?(@media_file)
    ffprobe = Settings&.ffprobe&.path || 'ffprobe'
    raw_output = `#{ffprobe} -i "#{@media_file.location}" -v quiet -show_format -show_streams -of json`
    # $? is a variable for the exit status of the last executed process. 
    # Success == 0, any other value means the command failed in some way.
    unless $?.exitstatus == 0
      Rails.logger.error "File processing failed. Please ensure that FFprobe is installed and that the correct path is configured."
      return @json_output = {}
    end
    @json_output = JSON.parse(raw_output).deep_symbolize_keys
    @json_output
  end

  def video_stream
    @video_stream ||= json_output[:streams].select { |stream| stream[:codec_type] == 'video' }.first
  end

  def audio_stream
    @audio_stream ||= json_output[:streams].select { |stream| stream[:codec_type] == 'audio' }.first
  end
```